### DISCUSSION TITLE
Update firmware with .bin file

### DISCUSSION THREAD

**Discussion by dwang-git** [semuconsulting/PyGPSClient#104] [General]
I'm using the u-center to update the UBlox firmware with .bin file. I suppose I can do this with PyGPSClient as well. Should I just treat the  .bin file just like .txt file with this tool? Thanks!

**Comment by semuadmin** [comment]
Hi @dwang-git 

No, unfortunately you can't currently use PyGPSClient to update the firmware on u-blox devices. 

u-blox have not put the firmware upgrade protocols in the public domain - they're only available under Non-Disclosure Agreement (NDA) - so I'm not able to implement them in open source software.

Sorry about this, but it's out of my hands. PyGPSClient can only support public domain u-blox functionality.
